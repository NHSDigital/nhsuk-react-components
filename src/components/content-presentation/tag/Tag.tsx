import React, { FC, HTMLProps } from 'react';
import classNames from 'classnames';

interface TagProps extends HTMLProps<HTMLSpanElement> {
  color?:
    | 'white'
    | 'grey'
    | 'green'
    | 'aqua-green'
    | 'blue'
    | 'purple'
    | 'pink'
    | 'red'
    | 'orange'
    | 'yellow';
}

const Tag: FC<TagProps> = ({ className, color, ...rest }) => (
  <strong
    className={classNames('nhsuk-tag', { [`nhsuk-tag--${color}`]: color }, className)}
    {...rest}
  />
);

export default Tag;
