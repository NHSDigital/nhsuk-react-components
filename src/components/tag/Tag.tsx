import React, { HTMLProps } from 'react';
import classNames from 'classnames';

type TagColours =
  | 'white'
  | 'grey'
  | 'green'
  | 'aqua-green'
  | 'blue'
  | 'purple'
  | 'pink'
  | 'red'
  | 'orange'
  | 'yellow';

export interface TagProps extends HTMLProps<HTMLSpanElement> {
  color?: TagColours;
}

const Tag: React.FC<TagProps> = ({ className, color, ...rest }) => (
  <span
    className={classNames('nhsuk-tag', { [`nhsuk-tag--${color}`]: color }, className)}
    {...rest}
  />
);

export default Tag;
