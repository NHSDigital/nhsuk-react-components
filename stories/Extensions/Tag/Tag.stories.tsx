import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import Tag from '@components/content-presentation/tag/Tag';

const meta: Meta<typeof Tag> = {
  title: 'Extensions/Tags',
  component: Tag,
};
export default meta;

type Story = StoryObj<typeof Tag>;

export const Basic: Story = {
  render: () => (
    <div className="tag-demo">
      <Tag>Standard</Tag>
      <Tag>Done</Tag>
      <Tag color="white">Started</Tag>
      <Tag color="grey">Not Started</Tag>
      <Tag color="blue">Ready to submit</Tag>
      <Tag color="red">FP69</Tag>
      <Tag color="orange">Ceased - no cervix</Tag>
    </div>
  ),
};

export const Colours: Story = {
  render: () => (
    <div className="tag-demo">
      <Tag>Standard</Tag>
      <Tag color="white">Started</Tag>
      <Tag color="grey">Not Started</Tag>
      <Tag color="green">New</Tag>
      <Tag color="aqua-green">Active</Tag>
      <Tag color="blue">Pending</Tag>
      <Tag color="purple">Received</Tag>
      <Tag color="pink">Sent</Tag>
      <Tag color="red">Rejected</Tag>
      <Tag color="orange">Declined</Tag>
      <Tag color="yellow">Delayed</Tag>
    </div>
  ),
};
