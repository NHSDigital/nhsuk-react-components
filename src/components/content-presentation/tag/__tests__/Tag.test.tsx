import React, { ComponentProps } from 'react';
import { render } from '@testing-library/react';
import Tag from '../Tag';

describe('Tag', () => {
  it('matches snapshot', () => {
    const { container } = render(<Tag>Active</Tag>);

    expect(container).toMatchSnapshot();
  });

  it('renders a nhsuk-tag class', () => {
    const { container } = render(<Tag>Active</Tag>);

    expect(container.querySelector('strong.nhsuk-tag')).toBeTruthy();
  });

  it.each<ComponentProps<typeof Tag>['color']>([
    'white',
    'grey',
    'green',
    'aqua-green',
    'blue',
    'purple',
    'pink',
    'red',
    'orange',
    'yellow',
  ])('adds colour class %s ', (colour) => {
    const { container } = render(<Tag color={colour} />);

    expect(container.querySelector(`strong.nhsuk-tag.nhsuk-tag--${colour}`)).toBeTruthy();
  });
});
