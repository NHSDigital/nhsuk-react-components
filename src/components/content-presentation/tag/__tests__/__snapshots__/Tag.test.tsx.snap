// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches snapshot 1`] = `
<div>
  <strong
    class="nhsuk-tag"
  >
    Active
  </strong>
</div>
`;
